docs/src/main/asciidoc/hibernate-reactive.adoc

```diff
@@ -0,0 +1,229 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Using Hibernate Reactive
+
+include::./attributes.adoc[]
+:config-file: application.properties
+:reactive-doc-url-prefix: http://hibernate.org/reactive/documentation/1.1/reference/html_single/#getting-started
+
+link:https://hibernate.org/reactive/[Hibernate Reactive] is a reactive API for Hibernate ORM, supporting non-blocking database drivers
+and a reactive style of interaction with the database.
+
+[NOTE]
+----
+Hibernate Reactive works with the same annotations and most of the configuration described in the
+xref:quarkus-hibernate-orm.adoc[Hibernate ORM guide]. This guide will only focus on what's specific
+for Hibernate Reactive
+----
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `hibernate-reactive-quickstart` {quickstarts-tree-url}/hibernate-reactive-quickstart[directory].
+
+[[hr-getting-started]]
+== Setting up and configuring Hibernate Reactive
+
+When using Hibernate Reactive in Quarkus, you need to:
+
+* add your configuration settings in `{config-file}`
+* annotate your entities with `@Entity` and any other mapping annotation as usual
+
+Other configuration needs have been automated: Quarkus will make some opinionated choices and educated guesses.
+
+Add the following dependencies to your project:
+
+* the Hibernate Reactive extension: `io.quarkus:quarkus-hibernate-reactive`
+* the Vert.x SQL client for the database of your choice; the following options are available:
+  - `quarkus-reactive-pg-client` for link:https://vertx.io/docs/vertx-pg-client/java[PostgreSQL or CockroachDB]
+  - `quarkus-reactive-mysql-client` for link:https://vertx.io/docs/vertx-mysql-client/java[MySQL or MariaDB]
+  - `quarkus-reactive-mssql-client` for link:https://vertx.io/docs/vertx-mssql-client/java[MS SQL Server]
+  - `quarkus-reactive-db2-client` for link:https://vertx.io/docs/vertx-db2-client/java[Db2]
+
```

Comment:
I think we have an Oracle driver now too.

Also I wonder if you should link to https://quarkus.io/guides/reactive-sql-clients instead of the Vert.x doc.

Reply:
You are right, I will change a bit this

Reply:
I've updated the section, let me know if it's better.

Reply:
AFAICS, you haven't added the Oracle one?

Also, there are other comments you didn't see: see https://github.com/quarkusio/quarkus/pull/22988/files for all of them.

Reply:
We haven't release the Oracle integration yet

Reply:
Ah, it requires some work in HR? I thought having the client around would be enough.

Reply:
I wish :)

Scaffolding apart, there is usually some custom logic needed because the reactive SQL clients don't follow a specific standard and sometimes they behave differently than what we expect.

At the moment, we are waiting for some fixes in Vert.x but I think I'm going to release it anyway next time and document the limitations.

Reply:
OK, thanks for the explanations!